tools/proguard5.3.2/test/proguard/util/ArrayUtilTest.java

```diff
@@ -0,0 +1,614 @@
+package proguard.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ArrayUtilTest {
+
+	@Test
+	public void testEqualByte() {
+		Assert.assertFalse(ArrayUtil.equal(
+				new byte[]{3, 2, 1}, new byte[]{1, 2, 3}, 4));
+
+		Assert.assertTrue(ArrayUtil.equal(new byte[]{1, 2, 3},
+				new byte[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new byte[]{1, 2, 3},
+				new byte[]{1, 2, 3, 4}, 3)
+		);
+	}
+
+	@Test
+	public void testEqualShort() {
+		Assert.assertFalse(ArrayUtil.equal(new short[]{3, 2, 1},
+				new short[]{1, 2, 3}, 4)
+		);
+
+		Assert.assertTrue(ArrayUtil.equal(new short[]{1, 2, 3},
+				new short[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new short[]{1, 2, 3},
+				new short[]{1, 2, 3, 4}, 3)
+		);
+	}
+
+	@Test
+	public void testEqualInt() {
+		Assert.assertFalse(ArrayUtil.equal(new int[]{3, 2, 1},
+				new int[]{1, 2, 3}, 4)
+		);
+
+		Assert.assertTrue(ArrayUtil.equal(new int[]{1, 2, 3},
+				new int[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new int[]{1, 2, 3},
+				new int[]{1, 2, 3, 4}, 3)
+		);
+	}
+
+	@Test
+	public void testEqualObject() {
+		Assert.assertFalse(ArrayUtil.equal(new Object[]{3, 2, 1},
+				new Object[]{1, 2, 3}, 4)
+		);
+
+		Assert.assertTrue(ArrayUtil.equal(new Object[]{1, 2, 3},
+				new Object[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new Object[]{1, 2, 3},
+				new Object[]{1, 2, 3, 4}, 3)
+		);
+	}
+
+	@Test
+	public void testEqualOrNull() {
+		Assert.assertTrue(ArrayUtil.equalOrNull(new Object[]{1, 2, 3},
+				new Object[]{1, 2, 3, 4})
+		);
+		Assert.assertTrue(ArrayUtil.equalOrNull(null, null));
```

Comment:
Move this to the top as it is only one line.. The split between true and false looks better then